lib/phlex/context.rb

```diff
@@ -68,9 +68,24 @@ def _void_element(**kwargs)
     end
 
     def _attributes(attributes)
+      if (cached = Phlex::ATTRIBUTE_CACHE[attributes.hash])
+        return @_target << cached
+      end
+
+      first_render = !self.class.rendered_at_least_once
```

Comment:
Maybe we could rename `self.class.rendered_at_least_once` into `self.class.first_render` to avoid the assignment to the variable and also have one single naming as we only use this here. What do you think?

Reply:
The assignment to variable makes it faster because local variables are faster than multiple method calls. The variable is named positively on the class because the default value of `nil` is falsy. If `self.class.first_render` is `nil`, that's falsy but it should be truth.